projects/plugins/paypal-payment-buttons/src/class-paypal-payment-buttons.php

```diff
@@ -0,0 +1,162 @@
+
+		// Provide block availability data for editor
+		add_action( 'enqueue_block_editor_assets', array( $this, 'enqueue_block_availability_data' ), 10 );
+
+		// Load styles in the editor iframe context
+		if ( is_admin() ) {
+			add_action( 'enqueue_block_assets', array( 'Automattic\Jetpack\PaypalPayments\PayPal_Payment_Buttons', 'load_editor_styles' ), 9 );
+		}
+
+		// Add admin menu
+		add_action( 'admin_menu', array( $this, 'add_admin_menu' ) );
+	}
+
+	/**
+	 * Register the PayPal Payment Buttons block with the correct dist path.
+	 */
+	public function register_paypal_block() {
+		// Get the path to the dist directory in the paypal-payments package
+		$package_dir = dirname( __DIR__ ) . '/vendor/automattic/jetpack-paypal-payments';
+		$dist_dir    = $package_dir . '/dist/paypal-payment-buttons';
+
+		if ( ! is_dir( $dist_dir ) ) {
+			return false;
+		}
+
+		// Register the block using the Blocks package with the correct dist path
+		\Automattic\Jetpack\Blocks::jetpack_register_block(
+			$dist_dir,
+			array(
+				'render_callback' => array( 'Automattic\Jetpack\PaypalPayments\PayPal_Payment_Buttons', 'render_block' ),
+			)
+		);
+	}
+
+	/**
+	 * Enqueue block availability data for the editor.
+	 */
+	public function enqueue_block_availability_data() {
+		// Only enqueue in the block editor
+		if ( ! function_exists( 'get_current_screen' ) ) {
+			return;
+		}
+
+		$screen = get_current_screen();
+		if ( ! $screen || ! $screen->is_block_editor() ) {
+			return;
+		}
+
+		// Provide the availability data that the block registration JavaScript expects
+		$availability_data = array(
+			'available_blocks' => array(
+				'paypal-payment-buttons' => array(
+					'available' => true,
+				),
+			),
+		);
+
+		wp_localize_script(
+			'jp-paypal-payments-ncps-blocks',
+			'Jetpack_Editor_Initial_State',
+			$availability_data
+		);
+	}
+
+	/**
+	 * Add admin menu for the plugin.
+	 */
+	public function add_admin_menu() {
+		$page_suffix = add_menu_page(
+			__( 'PayPal Payment Buttons', 'paypal-payment-buttons' ),
+			_x( 'PayPal Buttons', 'The PayPal Payment Buttons product name, without the Jetpack prefix', 'paypal-payment-buttons' ),
+			'manage_options',
+			'paypal-payment-buttons',
+			array( $this, 'plugin_settings_page' ),
+			'dashicons-money-alt'
+		);
+		add_action( 'load-' . $page_suffix, array( $this, 'admin_init' ) );
+	}
+
+	/**
+	 * Initialize the admin resources.
+	 */
+	public function admin_init() {
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_admin_scripts' ) );
+	}
+
+	/**
+	 * Main plugin settings page.
+	 */
+	public function plugin_settings_page() {
+		?>
+		<div id="paypal-payment-buttons-root"></div>
+		<?php
+	}
+
+	/**
+	 * Enqueue plugin admin scripts and styles.
+	 */
+	public function enqueue_admin_scripts() {
+		// Only enqueue on our admin page
+		$screen = get_current_screen();
+		if ( $screen && $screen->id !== 'toplevel_page_paypal-payment-buttons' ) {
+			return;
+		}
```

Comment:
This `enqueue_admin_scripts` is hooked into `admin_enqueue_scripts`, we could use `$hook_suffix`.


```suggestion
	 * @param string $hook_suffix The current admin page.
	 */
	public function enqueue_admin_scripts( $hook_suffix ) {
		// Only enqueue on our admin page
		if ( 'toplevel_page_paypal-payment-buttons' !== $hook_suffix ) {
			return;
		}
```